src/StmtToHTML.cpp

```diff
@@ -555,28 +555,40 @@ class AssemblyInfo : public IRVisitor {
         m.get_conceptual_stmt().accept(this);
     }
 
-    void generate(const std::string &code) {
+    void generate(std::string_view code) {
         // Find markers in asm code
-        std::istringstream asm_stream(code);
-        std::string line;
         int lno = 1;
-        while (getline(asm_stream, line)) {
-            // Try all markers
-            std::vector<uint64_t> matched_nodes;
-            for (auto const &[node, marker] : markers) {
-                if (std::regex_search(line, marker)) {
-                    // Save line number
-                    lnos[node] = lno;
-                    // Save this node's id
-                    matched_nodes.push_back(node);
-                }
+        size_t start = 0;
+
+        std::vector<uint64_t> matched_nodes;
+        while (start < code.size()) {
+            size_t end = code.find('\n', start);
+            if (end == std::string_view::npos) {
+                end = code.size();
             }
-            // We map to the first match, stop
-            // checking matched nodes
-            for (auto const &node : matched_nodes) {
-                markers.erase(node);
+            std::string_view line = code.substr(start, end - start);
+            std::string_view marker_prefix("%\"", 2);
+
+            // Quick check if the line contains %".
+            if (line.find(marker_prefix) != std::string_view::npos) {
+                // Try all markers
+                matched_nodes.clear();
+                for (auto const &[node, marker] : markers) {
+                    if (std::regex_search(line.begin(), line.end(), marker)) {
+                        // Save line number
+                        lnos[node] = lno;
+                        // Save this node's id
+                        matched_nodes.push_back(node);
+                    }
+                }
+                // We map to the first match, stop
+                // checking matched nodes
+                for (auto const &node : matched_nodes) {
+                    markers.erase(node);
+                }
             }
 
+            start = end + 1;
```

Comment:
The fact that start can ever point past the end of the string_view made me uneasy. I can't wait for `std::views::split` to be available (c++23)

Reply:
Well the loop condition should put you at ease. But indeed, a well-tested standard function would be nice.

Reply:
> Well the loop condition should put you at ease.

It does 🙂